README.md

```diff
@@ -46,5 +46,32 @@ to start the server in secure mode (both HTTPS and WSS)
 For local testing, use 'albano' as passphrase. Before you can use WSS with a self-signed certificate, you need to browse to https://localhost:4444 and add it to the certificate exceptions of your browser.
 
 Now you can navigate to localhost:8000 (https://localhost:4443 if https)
+
 and submit a job as professor
 
+
+# File downloads
+
+The flask server allows to use special tokens for file downloads.
+
+Example: to create the token for file alabno/tmp/test.txt
+run script alabno/frontend/CreateDownloadToken with argument
+alabno/tmp/test.txt
+CreateDownloadToken will print to stdout the token that can be used to retrieve the file.
+
+It's also possible to write the token to file with
+```
+./CreateDownloadToken alabno/tmp/test.txt --out /tmp/token.txt
+```
+
+To give users access to the file, use the link
+```
+http://localhost:8000/result/token
```

Comment:
because normally anti-hijacking is based on refreshing session cookies often, this would be fairly robust for 2 reasons:
* the token is temporary. While now it's set to 20 minutes for testing purposes, we will set to less than 1 minute in production, because the link will be generated instantly and consumed almost instantly by the user
* the user receiving the token for the download is getting this through websocket. The websocket connection is persistent and is not hijackable, also we don't have a system for 'remembering login' yet so it's really quite safe.